src/ray/raylet/node_manager.cc

```diff
@@ -539,6 +545,66 @@ ray::Status NodeManager::RegisterGcs() {
         "NodeManager.deadline_timer.print_event_loop_stats");
   }
 
+  if (RayConfig::instance().use_ray_syncer()) {
+    // Register resource manager and scheduler
+    ray_syncer_.Register(
+        syncer::RayComponentId::RESOURCE_MANAGER,
+        this,
+        this,
+        false,
+        RayConfig::instance().raylet_report_resources_period_milliseconds());
+    ray_syncer_.Register(syncer::RayComponentId::SCHEDULER,
+                         this,
+                         nullptr,
+                         true,
+                         RayConfig::instance().raylet_report_loads_period_milliseconds());
+    // Register a commands channel.
+    // It's only used for GC right now.
+    ray_syncer_.Register(syncer::RayComponentId::COMMANDS, nullptr, this, false, 0);
+    periodical_runner_.RunFnPeriodically(
+        [this] {
+          // If plasma store is under high pressure, we should try to schedule a global
+          // gc.
+          bool plasma_high_pressure =
```

Comment:
I assume this part of code is just copy/pasted, is this correct? (should I review it?)

Reply:
No. Hmmm, maybe I should refactor this a little bit. Let me check it in the interating.